lib/OpenQA/BuildResults.pm

```diff
@@ -85,13 +85,21 @@ sub add_review_badge {
 }
 
 sub compute_build_results {
-    my ($group, $limit, $time_limit_days, $tags) = @_;
+    my ($group, $limit, $time_limit_days, $tags, $subgroups) = @_;
 
     my $group_ids;
     my @children;
     if ($group->can('children')) {
-        @children  = $group->children;
         $group_ids = $group->child_group_ids;
+        if (@$subgroups) {
+            # filter subgroups
+            for my $child ($group->children) {
+                push(@children, $child) if grep { $_ eq '' || $child->name =~ /$_/ } @$subgroups;
```

Comment:
Unless subgroups are actually regular expressions, i think this is better written as `$child->name eq $_` or `$child->name =~ /\Q$_\E/`.

Reply:
I suppose it should actually support regex, similar to the filter for top-level groups we already have.